packages/core/src/search/details.ts

```diff
@@ -251,12 +254,20 @@ export function getExpressionsForResourceType(resourceType: string, expression:
   return result;
 }
 
-export function getExpressionForResourceType(resourceType: string, expression: string): string | undefined {
-  const atoms = getExpressionsForResourceType(resourceType, expression);
+export function getParsedExpressionForResourceType(resourceType: string, expression: string): FhirPathAtom {
+  const atoms: Atom[] = [];
+  const fhirPathExpression = parseFhirPath(expression);
+  buildExpressionsForResourceType(resourceType, fhirPathExpression.child, atoms);
+
   if (atoms.length === 0) {
-    return undefined;
+    return fhirPathExpression;
+  }
+
+  let result: Atom = atoms[0];
+  for (let i = 1; i < atoms.length; i++) {
+    result = new UnionAtom(result, atoms[i]);
   }
-  return atoms.map((atom) => atom.toString()).join(' | ');
+  return new FhirPathAtom('<original-not-available>', result);
```

Comment:
We could use `result.toString()` instead of a placeholder value, but that feels a little too self-referential